vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,438 @@
+
+```{r, eval = FALSE}
+adsl %>%
+  xportr_type(var_spec, "ADSL", "message") %>%
+  xportr_length(var_spec, "ADSL", "message") %>%
+  xportr_label(var_spec, "ADSL", "message") %>%
+  xportr_order(var_spec, "ADSL", "message") %>%
+  xportr_format(var_spec, "ADSL", "message") %>%
+  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+```
+
+To help reduce these repetitive calls, we have created `xportr_metadata()`. A user can just **set** the _metadata object_ and the Domain name in the first call, and this will be passed on to the other functions.  Much cleaner!
+
+
+```{r, eval = FALSE}
+adsl %>%
+  xportr_metadata(var_spec, "ADSL") %>%
+  xportr_type() %>%
+  xportr_length() %>%
+  xportr_label() %>%
+  xportr_order() %>%
+  xportr_format() %>%
+  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+```
+
+
+## Warnings and Errors 
+
+For the next six sections, we are going to explore the Warnings and Errors messages generated by the `{xportr}` core functions. To better explore these, we will either manipulate the ADaM dataset or specification file to help showcase the ability of the `{xportr}` functions to detect issues.  
+
+**NOTE:** We have made the ADSL, `xportr::adsl`, and Specification File, `xportr::var_spec`, available in this package. Users can find additional datasets and specification files on our [repo](https://github.com/atorus-research/xportr) in the `example_data_specs` folder. 
+
+### Setting up our metadata object
+
+First, let's read in the specification file and call it `var_spec`.  Note that we are not using `options()` here.  We will do some slight manipulation to the column names by doing all lower case, and changing `Data Type` to `type` and making the Order column numeric. You can also use `options()` for this step as well.  The `var_spec` object has five dataset specification files stacked on top of each other. We will make use of the `ADSL` subset of `var_spec`. You can make use of the Search field above the dataset column to subset the specification file for `ADSL`
+
+```{r}
+var_spec <- var_spec %>%
+  rename(type = "Data Type") %>%
+  set_names(tolower)
+```
+
+```{r, echo = FALSE}
+columns2hide <- c(
+  "significant digits", "mandatory", "assigned value", "codelist", "common",
+  "origin", "pages", "method", "predecessor", "role", "comment",
+  "developer notes"
+)
+
+datatable(
+  var_spec %>% select(-all_of(columns2hide)),
+  rownames = FALSE,
+  filter = "top",
+  options = list(
+    dom = "Bfrtip",
+    columnDefs = list(
+      list(width = "10px", targets = c("order", "length", "format", "type", "dataset", "variable")),
+      list(className = "text-left", targets = c("label"))
+    ),
+    searching = FALSE,
+    autoWidth = TRUE
+  )
+) %>%
+  formatStyle(0,
+    target = "row", color = "black", backgroundColor = "white",
+    fontWeight = "bold", lineHeight = "70%", textAlign = "center"
```

Comment:
Bold font and line height looks weird. I suggest a middle ground on the weight between normal and bold.

```suggestion
    fontWeight = "500", lineHeight = "85%", textAlign = "center"
```

![image](https://github.com/atorus-research/xportr/assets/211358/ef1dc5d9-dcad-4335-81ca-f0118e92121a)